### PR TITLE
NAD-1093 docs(textarea): add design ref tab

### PR DESCRIPTION
**Pull request by kevinch**
#### What is the purpose of this pull request?
Add design ref tab.

#### What problem is this solving?
Lack of content.

#### How should this be manually tested?
See links below.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-docs/docs/guidelines/components/text-area.mdx [hunk 1]
```diff
@@ -125,3 +148,18 @@ It also accepts all props of `textarea` JSX element.
...
+</Panel>
+<Panel id="designReference">
+
+## Do’s
+
+1. Use the Helper Text to explain in more details how the input is expected to be filled.
+2. If extra context is needed to clarify to the user where, or how the text inputted will appear, add a tooltip.
+3. Add placeholders to clarify the type of content that must be inserted by the user.
+4. Do use the required/optional flags when necessary, but prefer using one or another.
5. Add data validation to indicate when data hasn’t been inputted correctly.
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/guidelines/components/text-area.mdx:160]: Missing a line break

File: packages/admin-ui-docs/docs/guidelines/components/text-area.mdx [hunk 2]
```diff
@@ -125,3 +148,18 @@ It also accepts all props of `textarea` JSX element.
...
 | tone         | `neutral, critical`                           | TextArea's tone of voice                     | 🚫       | `neutral` |
 | csx          | `StyleProp`                                   | Defines component styles                     | 🚫       | `{}`      |
+
+</Panel>
+<Panel id="designReference">
+
+## Do’s
+
+1. Use the Helper Text to explain in more details how the input is expected to be filled.
+2. If extra context is needed to clarify to the user where, or how the text inputted will appear, add a tooltip.
```
Comment (davicostalf) [inline packages/admin-ui-docs/docs/guidelines/components/text-area.mdx:158]: Please remove this second point

**Review by davicostalf — CHANGES_REQUESTED** [review]

File: packages/admin-ui-docs/docs/guidelines/components/text-area.mdx (hunk 1 above)
Comment (marcelovicentegc) [reply packages/admin-ui-docs/docs/guidelines/components/text-area.mdx:160]: Done

File: packages/admin-ui-docs/docs/guidelines/components/text-area.mdx (hunk 2 above)
Comment (marcelovicentegc) [reply packages/admin-ui-docs/docs/guidelines/components/text-area.mdx:158]: Done

**Review by marcelovicentegc — APPROVED** [review]

**Pull request merged**